drivers/eeprom/eeprom_st25dv.c

```diff
@@ -0,0 +1,970 @@
+	ST25DV_REG_RFA1SS = 0x0004,
+	ST25DV_REG_ENDA1 = 0x0005,
+	ST25DV_REG_RFA2SS = 0x0006,
+	ST25DV_REG_ENDA2 = 0x0007,
+	ST25DV_REG_RFA3SS = 0x0008,
+	ST25DV_REG_ENDA3 = 0x0009,
+	ST25DV_REG_RFA4SS = 0x000a,
+	ST25DV_REG_I2CSS = 0x000b,
+	ST25DV_REG_LOCK_CC = 0x000c,
+	ST25DV_REG_MB_MODE = 0x000d,
+	ST25DV_REG_MB_WDG = 0x000e,
+	ST25DV_REG_LOCK_CFG = 0x000f,
+	ST25DV_REG_LOCK_DSF_ID = 0x0010,
+	ST25DV_REG_LOCK_AFI = 0x0011,
+	ST25DV_REG_DSFID = 0x0012,
+	ST25DV_REG_AFI = 0x0013,
+	ST25DV_REG_MEM_SIZE = 0x0014,
+	ST25DV_REG_BLK_SIZE = 0x0016,
+	ST25DV_REG_IC_REF = 0x0017,
+	ST25DV_REG_UUID = 0x0018,
+	ST25DV_REG_IC_REV = 0x0020,
+	ST25DV_REG_I2C_PASSWD = 0x0900,
+	ST25DV_REG_I2C_SSO_DYN = 0x2004,
+} st25dv_reg_t;
+
+/** @brief Length of the MEM_SIZE field of the ST25DV */
+#define ST25DV_MEM_SIZE_LENGTH   2u
+/** @brief Length of the BLK_SIZE field of the ST25DV */
+#define ST25DV_BLK_SIZE_LENGTH   1u
+/** @brief Length of the IC_REV field of the ST25DV */
+#define ST25DV_IC_REV_LENGTH     1u
+/** @brief Length of ENDA registers */
+#define ST25DV_ENDA_LENGTH       1u
+/** @brief Length of I2C_PASSWD message */
+#define ST25DV_I2C_PASSWD_LENGTH 17u
+/** @brief Length of RFAxSS registers */
+#define ST25DV_RFASS_LENGTH      1u
+
+/**
+ * @brief I2C write timeout (ms)
+ *
+ * @details Max value: t_write + t_write * (max write bytes) / (internal page write)
+ * 5 + 5*(256/16)
+ */
+#define EEPROM_ST25DV_WRITE_TIMEOUT 85u
+
+/* I2C Security Session Open I2C_SSO_Dyn */
+#define EEPROM_ST25DV_I2C_SSO_DYN_I2CSSO_SHIFT 0
+#define EEPROM_ST25DV_I2C_SSO_DYN_I2CSSO_FIELD 0xFE
+#define EEPROM_ST25DV_I2C_SSO_DYN_I2CSSO_MASK  0x01
+
+/* RF X Access Protection RFAxSS */
+#define EEPROM_ST25DV_RFAXSS_PWD_CTRL_SHIFT 0
+#define EEPROM_ST25DV_RFAXSS_PWD_CTRL_FIELD 0xFC
+#define EEPROM_ST25DV_RFAXSS_PWD_CTRL_MASK  0x03
+#define EEPROM_ST25DV_RFAXSS_RW_PROT_SHIFT  2u
+#define EEPROM_ST25DV_RFAXSS_RW_PROT_FIELD  0xF3
+#define EEPROM_ST25DV_RFAXSS_RW_PROT_MASK   0x0C
+#define EEPROM_ST25DV_RFAXSS_ALL_SHIFT      0
+#define EEPROM_ST25DV_RFAXSS_ALL_MASK       0x0F
+
+/* I2C Access Protection I2CSS */
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A1_SHIFT 0
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A1_FIELD 0xFC
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A1_MASK  0x03
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A2_SHIFT 2u
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A2_FIELD 0xF3
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A2_MASK  0x0C
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A3_SHIFT 4u
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A3_FIELD 0xCF
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A3_MASK  0x30
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A4_SHIFT 6u
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A4_FIELD 0x3F
+#define EEPROM_ST25DV_I2CSS_RW_PROT_A4_MASK  0xC0
+
+struct eeprom_st25dv_config {
+	struct i2c_dt_spec bus;
+	uint16_t addr;
+	uint8_t power_state;
+	size_t size;
+};
+
+struct eeprom_st25dv_data {
+	struct k_mutex lock;
+};
+
+static int _st25dv_read(const struct eeprom_st25dv_config *config, uint16_t addr, off_t offset,
```

Comment:
Please use the same coding style as in other drivers. E.g.:

```suggestion
static int eeprom_st25dv_read(const struct eeprom_st25dv_config *config, uint16_t addr, off_t offset,
```

Reply:
_st25dv_read is a private function internal to the driver. I already have a eeprom_st25dv_read function that is used for the EEPROM api

```
static const struct eeprom_driver_api eeprom_st25dv_api = {
	.read = eeprom_st25dv_read,
	.write = eeprom_st25dv_write,
	.size = eeprom_st25dv_size,
};
```